be/src/exec/json_scanner.cpp

```diff
@@ -0,0 +1,546 @@
+            }
+            _total_lines = get_data_by_jsonpath(slot_descs);
+            if (_total_lines == -1) {
+                return Status::InternalError("Parse json data is failed.");
+            } else if (_total_lines == 0) {
+                *eof = true;
+                return Status::OK();
+            }
+            _next_line = 0;
+        }
+
+        RETURN_IF_ERROR(set_tuple_value_from_map(tuple, slot_descs, tuple_pool, &valid));
+        _next_line++;
+        if (!valid) {// read a invalid row, then read next one
+            continue;
+        }
+        break; // read a valid row, then break
+    } while (_next_line <= _total_lines);
+    return Status::OK();
+}
+
+/**
+ * flat array for json
+ * For example:
+ *  [{"colunm1":"value1", "colunm2":10}, {"colunm1":"value2", "colunm2":30}]
+ * Result:
+ *      colunm1    colunm2
+ *      ------------------
+ *      value1     10
+ *      value2     30
+ */
+Status JsonReader::handle_flat_array_complex_json(Tuple* tuple, const std::vector<SlotDescriptor*>& slot_descs, MemPool* tuple_pool, bool* eof) {
+    do {
+        if (_next_line >= _total_lines) {//parse json and generic document
+            RETURN_IF_ERROR(parse_json_doc(eof));
+            if (*eof) {// read all data, then return
+                return Status::OK();
+            }
+            _total_lines = _json_doc.Size();
+            _next_line = 0;
+        }
+        int nullcount = 0;
+        bool valid = true;
+        size_t limit = std::min(slot_descs.size(), (size_t)_jsonpaths_doc.Size());
+        rapidjson::Value& objectValue = _json_doc[_next_line];
+
+        for (size_t i = 0; i < limit; i++) {
+            const rapidjson::Value& path = _jsonpaths_doc[i];
+            if (!path.IsString()) {
+                return Status::InternalError("Jsonpath is not string.");
+            }
+
+            // if jsonValues is null, because not match in jsondata.
+            rapidjson::Value* json_values = JsonFunctions::get_json_object_from_parsed_json(path.GetString(), &objectValue, _json_doc.GetAllocator());
+            if (json_values == nullptr) {
+                if (slot_descs[i]->is_nullable()) {
+                    tuple->set_null(slot_descs[i]->null_indicator_offset());
+                    nullcount++;
+                } else  {
+                    std::stringstream str_error;
+                    str_error << "The column `" << slot_descs[i]->col_name() << "` is not nullable, but it's not found in jsondata.";
+                    _state->append_error_msg_to_file("", str_error.str());
+                    _counter->num_rows_filtered++;
+                    valid = false; // current row is invalid
+                    break;
+                }
+            } else {
+                RETURN_IF_ERROR(write_data_to_tuple(json_values, slot_descs[i], tuple, tuple_pool));
+            }
+        }
+        _next_line++;
+        if (!valid || nullcount == _jsonpaths_doc.Size()) {
```

Comment:
if `nullcount == _jsonpaths_doc.Size()`, the `_counter->num_rows_filtered++` should be called, and the error line should be recorded in runtime state.

Reply:
ok